components/ReleaseStats/ReleaseStats.jsx

```diff
@@ -0,0 +1,142 @@
+import { useState } from "react"
+import { createClientComponentClient } from "@supabase/auth-helpers-nextjs"
+import {
+  Dialog,
+  DialogTrigger,
+  DialogContent,
+  DialogPortal,
+  DialogClose,
+} from "@/components/Dialog/Dialog"
+
+let today = new Date()
+
+let yesterday = new Date(today)
+yesterday.setDate(yesterday.getDate() - 1)
+
+let tomorrow = new Date(today)
+tomorrow.setDate(tomorrow.getDate() + 1)
+
+let pastWeek = new Date(today)
+pastWeek.setDate(pastWeek.getDate() - 6)
+
+let pastTwoWeeks = new Date(today)
+pastTwoWeeks.setDate(pastTwoWeeks.getDate() - 13)
+
+let past30Days = new Date(today)
+past30Days.setDate(past30Days.getDate() - 29)
+
+let allDates = new Date("2023-01-22")
+
+export default function ReleaseStats({ release }) {
+  const [open, setOpen] = useState(false)
+  const [redeemedCodes, setRedeemedCodes] = useState()
+  const [labels, setLabels] = useState()
+  const [chartData, setChartData] = useState()
+  const supabase = createClientComponentClient()
+
+  async function getCodes(startDate, endDate) {
+    try {
+      let { count, error } = await supabase
+        .from("codes")
+        .select("redeemed_at", { count: "exact", head: true })
+        .eq("release_id", release.id)
+        .eq("redeemed", true)
+        .gte("redeemed_at", startDate)
+        .lte("redeemed_at", endDate)
+
+      if (count) {
+        setRedeemedCodes(count)
+      } else if (count === 0) {
+        setRedeemedCodes(0)
+      }
+
+      if (error) throw error
+    } catch (error) {
+      throw error
+    }
+  }
+
+  const statsTime = [
+    {
+      label: "Today",
+      start: today,
+      end: tomorrow,
```

Comment:
Why tomorrow? Shouldn't it just end right now, which is done when you declare `new Date()`? This feels unnecessary. You're trying to fetch data from the future which doesn't exist yet.